crates/holochain/tests/sharded_gossip/mod.rs

```diff
@@ -201,6 +201,53 @@ async fn test_gossip_shutdown() {
     assert_eq!(record.unwrap().action_address(), &hash);
 }
 
+/// Test that when a new conductor joins, gossip picks up existing data without needing a publish.
+#[cfg(feature = "slow_tests")]
+#[tokio::test(flavor = "multi_thread")]
+#[ignore = "This test is potentially useful but uses sleeps and has never failed.
+            Run it again in the future to see if it fails, and if so, rewrite it without sleeps."]
+async fn test_gossip_startup() {
+    holochain_trace::test_run().ok();
+    let config = || {
+        SweetConductorConfig::rendezvous().tune(|t| {
+            t.danger_gossip_recent_threshold_secs = 1;
+            t.default_rpc_single_timeout_ms = 3_000;
+            t.gossip_round_timeout_ms = 10_000;
+        })
+    };
+
+    let (dna_file, _, _) = SweetDnaFile::unique_from_inline_zomes(simple_crud_zome()).await;
+    let mk_conductor = || async {
+        let cfg = config();
+        assert!(cfg.network.unwrap().is_tx5());
+        let mut conductor =
+            SweetConductor::from_config_rendezvous(cfg, SweetLocalRendezvous::new().await).await;
+        // let mut conductor = SweetConductor::from_config(config()).await;
+        let app = conductor.setup_app("app", [&dna_file]).await.unwrap();
+        let cell = app.into_cells().pop().unwrap();
+        let zome = cell.zome(SweetInlineZomes::COORDINATOR);
+        (conductor, cell, zome)
+    };
+    let (conductor0, cell0, zome0) = mk_conductor().await;
+
+    // Create an entry before the conductors know about each other
+    let hash: ActionHash = conductor0
+        .call(&zome0, "create_string", "hi".to_string())
+        .await;
+
+    // Startup and do peer discovery
+    tokio::time::sleep(std::time::Duration::from_secs(5)).await;
+    let (conductor1, cell1, zome1) = mk_conductor().await;
+
+    // Wait a bit so that conductor 0 doesn't publish in the next step.
+    tokio::time::sleep(std::time::Duration::from_secs(5)).await;
+    SweetConductor::exchange_peer_info([&conductor0, &conductor1]).await;
+
+    consistency_60s([&cell0, &cell1]).await;
```

Comment:
this seems very long, shouldn't gossip start directly after new peers were discovered? is that why the test passed before the fix?

Reply:
gossip actually succeeds in less than 3 seconds so this timeout is not helping it pass